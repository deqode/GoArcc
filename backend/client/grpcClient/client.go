package grpcClient

import (
	"alfred/config"
	"alfred/logger"
	"alfred/servers/openTracing/tracer/jaeger"
	"context"
	grpc_opentracing "github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing"
	"go.uber.org/zap"
	"google.golang.org/grpc"
	"time"
)

//ClientContext to store context
type ClientContext struct {
	Ctx context.Context
}

//GetGrpcClientConnection is used to get the client connection
func GetGrpcClientConnection(config *config.Config) *grpc.ClientConn {
	var opts []grpc.DialOption
	opts = append(opts,
		grpc.WithUnaryInterceptor(
			//tracing for unary interceptors
			grpc_opentracing.UnaryClientInterceptor(grpc_opentracing.WithTracer(jaeger.Tracer)),
		),
		//tracing for stream interceptors
		grpc.WithStreamInterceptor(grpc_opentracing.StreamClientInterceptor(grpc_opentracing.WithTracer(jaeger.Tracer))),
	)

	//append grpc insecure
	opts = append(opts,
		grpc.WithInsecure(),
	)
	ctx, cancel := context.WithTimeout(context.Background(), time.Second*time.Duration(config.Grpc.RequestTimeout))
	defer cancel()

	// You must have some sort of OpenTracing Tracer instance on hand.
	conn, err := grpc.DialContext(ctx, config.Grpc.Host+":"+config.Grpc.Port, opts...)
	if err != nil {
		logger.Log.Fatal("did not connect", zap.Error(err))
	}
	return conn
}
