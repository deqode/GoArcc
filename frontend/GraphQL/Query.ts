import { gql } from '@apollo/client';

export const GET_REPOSITORIES = gql`
query repositories(
    $userid:String!
    $accountid:String!
    $provider:Types_Enum_GitProviders!
    ){
    repositories(
      provider:$provider,
      user_id:$userid,
      account_id:$accountid,
    ){repositories{
      name
    }}
  }
`
// todo : gql file

export const GET_BRANCHES = gql`
query repository(
    $ownerName:String!
    $repoName:String!
    $accountid:String!
    $provider:Types_Enum_GitProviders!
){
    repository(
      owner_name:$ownerName
      provider:$provider,
      repo_name:$repoName
      account_id:$accountid
    ){
      branches
    }
  }`

export const VCS_CONNECTIONS = gql`
query VCSConnections(
  $accountid:String!
  $provider:Types_Enum_GitProviders!
){
  VCSConnections(
    account_id:$accountid
    provider:$provider,
    ){
    vcs_connection{
            user_name
    }
  }
}`