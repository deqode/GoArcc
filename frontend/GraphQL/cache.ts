import { gql, InMemoryCache, makeVar } from '@apollo/client'

export const token = makeVar(null)

export default new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        token: {
          read() {
            return token()
          },
        },
      },
    },
  },
})

export const GET_TOKEN = gql`
  query getToken {
    token @client
  }
`
