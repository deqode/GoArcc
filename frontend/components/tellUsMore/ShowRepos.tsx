import { useQuery } from '@apollo/client'
import {
  CircularProgress,
  FormControl,
  Grid,
  InputLabel,
  MenuItem,
  Select,
} from '@material-ui/core'
import React, { Dispatch, ReactElement, SetStateAction, useEffect, useState } from 'react'
import { GET_REPOSITORIES } from '../../GraphQL/Query'
import { useSelectStyles } from '../../styles/commonStyles'

export default function ShowRepos({
  userId,
  accountId,
  setCurrentRepo,
  provider,
}: {
  userId: string
  accountId: string
  setCurrentRepo: Dispatch<SetStateAction<string>>
  provider: string
}): ReactElement {
  const { loading, error, data } = useQuery(GET_REPOSITORIES, {
    variables: {
      provider: provider,
      userId,
      accountId,
    },
  })

  const [repos, setRepos] = useState([])
  const classes = useSelectStyles()

  useEffect(() => {
    // if(error)
    // TODO:pop error
    if (data && data.repositories) setRepos(data.repositories.repositories || [])
  }, [data, error])

  const selectRepo = (
    e: React.ChangeEvent<{
      name?: string | undefined
      value: unknown
    }>
  ): void => {
    if (typeof e.target.value === 'string') setCurrentRepo(e.target.value)
    else setCurrentRepo('')
  }

  console.log(loading, error, data, "+++++++++++++++++++++++++++++++")

  return (
    <Grid item xs={12} md={12}>
      <FormControl variant="outlined">
        <InputLabel id="demo-simple-select-outlined-label">Repo Name</InputLabel>
        <Select
          className={classes.root}
          labelId="demo-simple-select-outlined-label"
          id="demo-simple-select-outlined"
          label="Repo Name"
          disabled={!(repos.length > 0)}
          onChange={selectRepo}>
          {loading && (
            <MenuItem value="">
              <CircularProgress color="secondary" />
            </MenuItem>
          )}
          {repos.map((r: { name: string }) => (
            <MenuItem key={r.name} value={r.name}>
              {r.name}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    </Grid>
  )
}
