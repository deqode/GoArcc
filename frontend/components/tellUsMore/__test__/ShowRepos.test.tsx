import TestRenderer from 'react-test-renderer'
import { MockedProvider } from '@apollo/client/testing'
import { waitFor } from '@testing-library/react'
import { GET_REPOSITORIES } from '../../../GraphQL/Query'
import ShowRepos from '../ShowRepos'

const mocks = [
  {
    request: {
      query: GET_REPOSITORIES,
      variables: {
        provider: 'provider',
        userId: 'userId',
        accountId: 'accountId',
      },
    },
    result: {
      data: {
        repositories: {
          repositories: [
            {
              name: 'REPO1',
            },
            {
              name: 'REPO2',
            },
          ],
        },
      },
    },
  },
]

const props = {
  userId: 'userId',
  accountId: 'accountId',
  setCurrentRepo: jest.fn(),
  provider: 'provider',
}
// TODO need to wait for load

it('should render repo list', async () => {
  let component: any
  const { act } = TestRenderer

  await new Promise((resolve) => setTimeout(resolve, 0))

  await act(async () => {
    component = TestRenderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ShowRepos {...props} />
      </MockedProvider>
    )
  })

  await waitFor(() => {
    // expect(p.children.join('')).toContain('Buck is a poodle')
    expect(component.toJSON()).toMatchSnapshot()
  })
})
