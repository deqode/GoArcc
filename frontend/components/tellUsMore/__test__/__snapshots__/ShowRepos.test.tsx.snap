// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render repo list 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
>
  <div
    className="MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      id="demo-simple-select-outlined-label"
    >
      Repo Name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-1 MuiInputBase-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        id="demo-simple-select-outlined"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-4"
        >
          <span>
            Repo Name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
